File: backend/app/schemas/report.py
from pydantic import BaseModel
from typing import List


class KPIResponse(BaseModel):
    total_patients: int = 0
    paid: int = 0
    partial: int = 0
    pending: int = 0
    today_patients: int = 0


class PatientTrendResponse(BaseModel):
    month: str
    patients: int


class ExpenseComparisonResponse(BaseModel):
    month: str
    expenses: float

File: frontend/types/report.ts
export interface KPIResponse {
  total_patients: number;
  paid: number;
  partial: number;
  pending: number;
  today_patients: number;
}

export interface PatientTrendResponse {
  month: string;
  patients: number;
}

export interface ExpenseComparisonResponse {
  month: string;
  expenses: number;
}
